beacon-chain/sync/subscriber.go

```diff
@@ -156,6 +156,19 @@ func (s *Service) registerSubscribers(epoch primitives.Epoch, digest [4]byte) {
 			func(currentSlot primitives.Slot) []uint64 { return []uint64{} },
 		)
 	}
+
+	if params.BeaconConfig().ElectraForkEpoch <= epoch {
```

Comment:
Instead of writing this entire new section, 


```go
func(primitives.Slot) []uint64 { return sliceFromCount(params.BeaconConfig().BlobsidecarSubnetCount) },
```

can be modified by a slight more complex function, that checks the epoch from the input slot, and returns either 
```go
sliceFromCount(params.BeaconConfig().BlobsidecarSubnetCount)
```

if we are in Deneb, or
```go
sliceFromCount(params.BeaconConfig().BlobsidecarSubnetCountElectra)
```

if we are in Electra.

(This function is executed at each slot.)

Reply:
I think this is what you meant, please correct if Im missunderstanding: https://github.com/prysmaticlabs/prysm/pull/14750/commits/4e62a244cd992e5c3e118c472b4ee43c3b21074b